### PR TITLE
Email on add book

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**

### CODE REVIEW

File: books/middleware/user_authentication_middleware.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from django.shortcuts import get_object_or_404, render
+
+from books.models import Book
+
+
+class UserAuthenticationMiddleware(object):
```
Comment (danialmalik) [inline books/middleware/user_authentication_middleware.py]: Please use decorator methods for such purposes.

File: books/migrations/0003_auto_20190729_1235.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.3 on 2019-07-29 12:35
+
+from django.db import migrations, models
+import django.db.models.deletion
```
Comment (danialmalik) [inline books/migrations/0003_auto_20190729_1235.py]: Please use better imports.

File: books/migrations/0004_auto_20190731_1005.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Generated by Django 2.2.3 on 2019-07-31 10:05
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
```
Comment (danialmalik) [inline books/migrations/0004_auto_20190731_1005.py]:
```suggestion
from django.db.models.deletion import CASCADE
```

File: books/migrations/0004_auto_20190731_1005.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('books', '0003_auto_20190729_1235'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='book',
+            name='user',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL),
```
Comment (danialmalik) [inline books/migrations/0004_auto_20190731_1005.py]:
```suggestion
            field=models.ForeignKey(on_delete=CASCADE, to=settings.AUTH_USER_MODEL),
```

File: books/models.py [hunk 5]
```diff
@@ -1,7 +1,49 @@
+import ssl
+
+from django.contrib.auth import get_user_model
 from django.db import models
+from django.db.models.signals import post_save
+import smtplib
+
+from django.dispatch import receiver
```
Comment (danialmalik) [inline books/models.py]: Please sort the imports

File: books/models.py [hunk 6]
```diff
@@ -1,7 +1,49 @@
...
     title = models.CharField(max_length=100)
     author = models.CharField(max_length=100)
     description = models.CharField(max_length=1000)
+    user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE)
+
+
+@receiver(post_save, sender=Book)
+def send_email(sender, instance, **kwargs):
+    port = 465  # For SSL
+    sender_email = "AliAkbarEdly@gmail.com"
```
Comment (danialmalik) [inline books/models.py]: Please don't use hard coded credentials

File: books/views.py [hunk 7]
```diff
@@ -1,6 +1,66 @@
...
+
+def update_book(request, pk):
+    if request.method == "POST":
+        book = get_object_or_404(Book, id=pk)
+        if book is not None:
+            book.title = request.POST.get('title')
+            book.author = request.POST.get('author')
+            book.description = request.POST.get('description')
+            book.save()
+            return redirect(reverse('books:detail', args=[pk]))
```
Comment (danialmalik) [inline books/views.py]: What will happen if `book` is `None` ? 
and why do you need the `if` condition if you are using `get_object_or_404` ?

File: books/middleware/user_authentication_middleware.py (hunk 1 above)
Comment (Ali-D-Akbar) [reply books/middleware/user_authentication_middleware.py]: this middleware was asked to be written.

File: books/views.py (hunk 7 above)
Comment (Ali-D-Akbar) [reply books/views.py]: the if condition was unnecessary. Removed it.

File: books/migrations/0003_auto_20190729_1235.py (hunk 2 above)
Comment (Ali-D-Akbar) [reply books/migrations/0003_auto_20190729_1235.py]: migrations are auto generated that's why I these files are like that

File: books/models.py (hunk 5 above)
Comment (Ali-D-Akbar) [reply books/models.py]: done

**Pull request merged**